io-functions-pay-portal/PostNewslettersRecipients/__tests__/handler.test.ts

```diff
@@ -0,0 +1,196 @@
+
+    const handler = handlers.PostNewslettersRecipientsHandler();
+
+    const response = await handler(
+      context,
+      groupId as NonEmptyString,
+      {
+        email,
+        recaptchaToken
+      } as RecipientRequest
+    );
+
+    expect(response.kind).toBe("IResponseSuccessJson");
+  });
+
+  it("should return IResponseErrorInternal if retrieval token mailup fails", async () => {
+    jest.spyOn(handlers, "recaptchaCheckTask").mockReturnValueOnce(
+      taskEither.of({
+        challenge_ts: "challenge_ts",
+        hostname: "hostname",
+        success: true
+      })
+    );
+
+    jest
+      .spyOn(handlers, "getMailupAuthTokenTask")
+      .mockReturnValueOnce(fromLeft(Error()));
+
+    const handler = handlers.PostNewslettersRecipientsHandler();
+
+    const response = await handler(
+      context,
+      groupId as NonEmptyString,
+      {
+        email,
+        recaptchaToken
+      } as RecipientRequest
+    );
+
+    expect(response.kind).toBe("IResponseErrorInternal");
+  });
+
+  it("should return IResponseErrorInternal if recaptchaCheck fails", async () => {
+    jest.spyOn(handlers, "recaptchaCheckTask").mockReturnValueOnce(
+      taskEither.of({
+        challenge_ts: "challenge_ts",
+        hostname: "hostname",
+        success: true
+      } as any)
+    );
+
+    jest
+      .spyOn(handlers, "recaptchaCheckTask")
+      .mockReturnValueOnce(fromLeft(Error()));
+
+    const handler = handlers.PostNewslettersRecipientsHandler();
+
+    const response = await handler(
+      context,
+      groupId as NonEmptyString,
+      {
+        email,
+        recaptchaToken
+      } as RecipientRequest
+    );
+
+    expect(response.kind).toBe("IResponseErrorInternal");
+  });
+
+  it("should return IResponseErrorInternal if adding receiver to mailup fails fails", async () => {
+    jest.spyOn(handlers, "recaptchaCheckTask").mockReturnValueOnce(
+      taskEither.of({
+        challenge_ts: "challenge_ts",
+        hostname: "hostname",
+        success: true
+      } as any)
+    );
+
+    jest.spyOn(handlers, "getMailupAuthTokenTask").mockReturnValueOnce(
+      taskEither.of({
+        access_token: "036AGdBq77634556uB8g-4s9SKbbg",
+        expires_in: 3600,
+        refresh_token: "03AGdBsd6ZV2qIuB8g-4s9SKbbg"
+      } as any)
+    );
+
+    jest
+      .spyOn(handlers, "addRecipientToMailupListTask")
+      .mockReturnValueOnce(fromLeft(Error()));
+
+    const handler = handlers.PostNewslettersRecipientsHandler();
+
+    const response = await handler(
+      context,
+      groupId as NonEmptyString,
+      {
+        email,
+        recaptchaToken
+      } as RecipientRequest
+    );
+
+    expect(response.kind).toBe("IResponseErrorInternal");
+  });
+
+  it("should return IResponseErrorForbiddenNotAuthorized if groupId is unauthorized", async () => {
+    jest
+      .spyOn(handlers, "addRecipientToMailupListTask")
+      .mockReturnValueOnce(fromLeft(Error()));
```

Comment:
I think this mock value is useless , `fromPredicate` test on `groupId` is the firs step in to handler

Reply:
right 👍